Tests/RoslynPluginGeneratorTests/AnalyzerPluginGeneratorTests.cs

```diff
@@ -87,6 +91,81 @@ public void Generate_LicenseAcceptanceNotRequired_Succeeds()
         }
 
         [TestMethod]
+        public void Generate_Recursive_Succeeds()
+        {
+            // Arrange
+            string outputDir = TestUtils.CreateTestDirectory(this.TestContext, ".out");
+            RemoteRepoBuilder remoteRepoBuilder = new RemoteRepoBuilder(this.TestContext);
+
+            // Multi-level dependencies: no package requires license acceptence
+            IPackage grandchild = CreatePackageWithAnalyzer(remoteRepoBuilder, "grandchild.id", "1.2", License.NotRequired /* no dependencies */);
+            IPackage child = CreatePackageWithAnalyzer(remoteRepoBuilder, "child.id", "1.1", License.NotRequired, grandchild);
+            IPackage parent = CreatePackageWithAnalyzer(remoteRepoBuilder, "parent.id", "1.0", License.NotRequired, child);
+
+            TestLogger logger = new TestLogger();
+            AnalyzerPluginGenerator apg = CreateTestSubjectWithFakeRemoteRepo(remoteRepoBuilder, logger);
+
+            // Act
+            ProcessedArgs args = CreateArgs("parent.id", "1.0", "cs", null, false, true /* /recurse = true */, outputDir);
+            bool result = apg.Generate(args);
+
+            // Assert
+            Assert.IsTrue(result, "Generator should succeed if there are no licenses to accept");
+            logger.AssertWarningNotLogged("parent.id"); // not expecting warnings about packages that don't require acceptance
+            logger.AssertWarningNotLogged("child.id");
+            logger.AssertWarningNotLogged("grandchild.id");
+            logger.AssertErrorsLogged(0);
+        }
+
+        [TestMethod]
+        public void Generate_LicenseAcceptanceNotRequired_NoAnalyzersInTarget()
```

Comment:
Keep related tests together (makes it easier to see that all of the various permutations are covered). This test looks like it's related to _Generate_LicenseAcceptanceRequested_NoAnalysersInTarget_ at line 314 below (although the test at line 314 should perhaps have _Required_ rather than _Requested_ in the title).